vincentvanbot/data.py

```diff
@@ -26,43 +24,79 @@ def get_data_locally(nrows=10):
 
 def download_images_locally(df):
     """Saves jpg files under raw_data/images based on paintings in df"""
+    tqdm.pandas(bar_format='{l_bar}{bar:30}{r_bar}{bar:-10b}')
+
     if not os.path.exists(IMAGES_PATH):
         os.mkdir(IMAGES_PATH)
     print(f'\nDownloading images to {IMAGES_PATH}...')
     df = df.progress_apply(download_single_image,axis=1)
 
-def get_joblib_data(size=100, source='gcp', rm=True):
-    """Gets joblib file from source and returns images df"""
-    # client = storage.Client()
+def create_flat_images_db(size=100, path=IMAGES_PATH, dim=(36,42)):
+    """For each image in path, resizes it to the given dim, transforms it into a flat vector
+    and stores in a df. Then dumps it into a joblib file"""
+
+    # stores flat images in a dataframe
+    img_db = pd.DataFrame()
+    print("\nCreating joblib database...")
+    for filename in tqdm(os.listdir(IMAGES_PATH)[:size], bar_format='{l_bar}{bar:30}{r_bar}{bar:-10b}'):
+        img = preprocess_image(os.path.join(IMAGES_PATH, filename),dim=dim)
+        img_db = img_db.append(pd.DataFrame(img,index=[filename.strip('.jpg')]))
+
+    # save df to joblib file
+    img_db.sort_index(inplace=True)
+    if size:
+        joblib.dump(img_db,FLAT_IMAGES_DB_PATH_ROOT+'_'+str(size)+'.joblib')
+    else:
+        joblib.dump(img_db,FLAT_IMAGES_DB_PATH_ROOT+'.joblib')
+
+def flat_images_db_upload(size=100, rm=False):
+    """Upload df of flat image vectors as joblib file to google cloud"""
+    client = storage.Client().bucket(BUCKET_NAME)
+
+    if size:
```

Comment:
Condition 'if size' will be always True, as you have a default value of 100, unless this method is called with size=0. Not sure if this is the assumed behavior.